src/controller.rs

```diff
@@ -114,27 +120,69 @@ async fn reconcile(app_instance: Arc<AppInstance>, ctx: Arc<Context>) -> Result<
         serde_json::from_value(config.metadata.get(PACK_KEY).unwrap().clone())
             .map_err(Error::DecodeKubecfgPackageMetadata)?;
 
-    launch_job(&app_instance, &kubecfg_pack_metadata, Arc::clone(&ctx)).await?;
+    let configmap_name = create_configmap_overlay(&app_instance, Arc::clone(&ctx)).await?;
+    launch_job(
+        &app_instance,
+        &kubecfg_pack_metadata,
+        configmap_name,
+        Arc::clone(&ctx),
+    )
+    .await?;
 
     Ok(Action::await_change())
 }
 
+fn calculate_hash<T: Hash>(t: &T) -> u64 {
+    let mut s = DefaultHasher::new();
+    t.hash(&mut s);
+    s.finish()
+}
```

Comment:
I think I could place this into `lib.rs`, since it would technically be "shared functionality", but it's also only going to be used by the `controller.rs` in its current form.

I've left it here for now, but happy to move it.